src/gherlint/checkers/convention.py

```diff
@@ -0,0 +1,79 @@
+import re
+from typing import Optional, Pattern, Union
+
+from pydantic import validator
+
+from gherlint.checkers.base_checker import BaseChecker
+from gherlint.objectmodel import nodes
+from gherlint.options import Options
+from gherlint.registry import CheckerRegistry
+from gherlint.reporting import Message
+
+
+class ConventionCheckerOptions(Options):
+    config_section = "convention"
+
+    feature_tags_pattern: Optional[Pattern]
+    scenario_tags_pattern: Optional[Pattern]
+
+    # pylint: disable=no-self-argument, disable=no-self-use
+    @validator("feature_tags_pattern", "scenario_tags_pattern", pre=True)
+    def compile_regex(cls, value: str) -> Optional[Pattern]:
+        if value:
+            return re.compile(value, re.X)
+
+        return None
+
+
+class ConventionChecker(BaseChecker):
+    options: ConventionCheckerOptions
+
+    MESSAGES = [
+        Message(
+            "C401",
+            "feature-tags-pattern-mismatch",
+            "Feature tags do not follow the pattern: {pattern}",
+        ),
+        Message(
+            "C402",
+            "scenario-tags-pattern-mismatch",
+            "Scenario tags do not follow the pattern: {pattern}",
+        ),
+    ]
+
+    def visit_feature(self, node: nodes.Feature) -> None:
+        if self.options.feature_tags_pattern:
+            self._check_node_tags_with_pattern(
+                node,
+                self.options.feature_tags_pattern,
+                "feature-tags-pattern-mismatch",
+            )
+
+    def visit_scenario(self, node: nodes.Scenario) -> None:
```

Comment:
I think we could reduce the code duplications by using `visit_tag` instead of `visit_feature`/`visit_scenario`/`visit_scenariooutline`, and inspecting the `node.parent` attribute to determine which patterns should be applied. 
What do you think about that?

Reply:
Hm, I'll try to do that, thanks.

Reply:
Hm, the main problem with this solution is that I can't check tags one by one, I need to match the whole tags string with a pattern, that's why it's not possible to check with `visit_tag`. Certainly I can store each tag with a feature / scenario in a dict and later use it to check, but it will not be different from the current solution. Besides I don't know how to implement errors like `You missed required tags: required_1, required_2`, because with regex lib I can't know which group is optional or required.

So, I think we can replace regex solution with a custom check and split it into separate required and optional checks, but for now I don't know how to do that, I have no ideas. We need to think about it, maybe the regex solution is not good enough as I thought.